src/high_scores.rs

```diff
@@ -64,10 +77,65 @@ fn append_update_comment(filename: &str, old_version: &str) -> Result<(), AnyErr
     Ok(())
 }
 
-fn update_version_number(filename: &str) -> Result<(), AnyErrorThreadSafe> {
-    let mut file = fs::OpenOptions::new().write(true).open(filename)?;
-    file.seek(SeekFrom::Start(HEADER_PREFIX.len() as u64))?;
-    file.write_all(VERSION.as_bytes())?;
+fn update_from_v4_or_older(filename: &str) -> Result<(), AnyErrorThreadSafe> {
+    let temp_file = tempfile::tempfile()?;
+    let mut writer = BufWriter::new(temp_file);
+
+    writer.write_all(HEADER_PREFIX.as_bytes())?;
+    writer.write_all(VERSION.as_bytes())?;
+    writer.write_all(b"\n")?;
+
+    let mut old_file = fs::OpenOptions::new().read(true).open(filename)?;
+    let mut lines = BufReader::new(&mut old_file).lines();
+    lines.next().ok_or("high scores file is empty")??;
+
+    let mut lineno = 1;
+    for line in lines {
+        lineno += 1;
+
+        let line = line?;
+        if line.trim().starts_with('#') {
+            writer.write_all(line.trim().as_bytes())?;
+        } else if !line.trim().is_empty() {
+            let split_error = || {
+                format!(
+                    "not enough tab-separated parts on line {} of high scores file",
+                    lineno
+                )
+            };
+
+            let mut parts = line.split('\t');
+            let mode_name = parts.next().ok_or_else(split_error)?;
+            let _ = parts.next().ok_or_else(split_error)?; // lobby id
```

Comment:
Instead of completely redesigning the format, maybe use this field for timestamp?  Downside is then we can't save both start and end timestamp.